### PR TITLE
Dockerize

### LINKED ISSUE #7: Dockerize for Scalability
I don't really know a lot about python but should dockerize the app for easier future scalability.

### PR DESCRIPTION
**Pull request by laurin**
Closes #7

### CODE REVIEW

**Comment by laurin** [comment]
@kinshukdua The `tweets.txt` is not supposed to be persisted, right?

**Comment by kinshukdua** [comment]
@laurin it is supposed to persist for a short while. I'm thinking of letting the tweets accumulate for a bit (say 30 minutes or an hour) and then be deleted. I'm currently handling this with a cronjob but the feature being part of the docker container would help a lot.

**Comment by laurin** [comment]
Ok, the problem is that the tweets will be appended to the existing files, so most of there are a lot of duplicates.
If the 100 tweets max returned by the Twitter API are not enough we could retrieve multiple pages or properly store the tweets with an id.

**Comment by kinshukdua** [comment]
You're right, we probably need to store the tweets with an id, to prevent duplicates. I think a database is an overkill for storing simple tweets. Instead lets just store the tweets with ids, text and link in a sorted order. Reading the file should barely take any time so we can just check the ids.

**Comment by DomiiBunn** [comment]
Try firebase, it's simple yet perfect for these solutions

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.10-slim@sha256:2bac43769ace90ebd3ad83e5392295e25dfc58e58543d3ab326c3330b505283d
```
Comment (DomiiBunn) [inline Dockerfile:1]: I just want to understand the reasoning why specify the whole docker sha? That would stop it from updating if a security patch comes out for the image...

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: "3"
+
+services:
+  server:
+    image: nginx:1.21.6
+    restart: unless-stopped
+    volumes:
+      - ./data/dist:/usr/share/nginx/html:ro
```
Comment (DomiiBunn) [inline docker/docker-compose.yaml:8]: Why not just a docker volume instead of dealing with absolute paths?

**Review by DomiiBunn — CHANGES_REQUESTED** [review]
It would also be usefull to set up a CI so the image is at least built automatically. I woud go further and set one up to auto deploy everything on updates to the main branch.

File: docker/docker-compose.yaml (hunk 2 above)
Comment (laurin) [reply docker/docker-compose.yaml:8]: Good point, I just normally do it that way because it makes backups more convenient.

File: Dockerfile (hunk 1 above)
Comment (laurin) [reply Dockerfile:1]: Yeah that's probably just matter of opinion, pinning the image with a hash allows for a reproducible build.

**Comment by laurin** [comment]
I think we should take the discussion regarding the storage of tweets to another issue

File: Dockerfile (hunk 1 above)
Comment (DomiiBunn) [reply Dockerfile:1]: By all means it is ^.^

File: docker/docker-compose.yaml (hunk 2 above)
Comment (DomiiBunn) [reply docker/docker-compose.yaml]: True. Whoever i don't think this should count as persistent data

**Pull request merged**